internal/ethereum/blocklistener.go

```diff
@@ -470,23 +518,28 @@ func (bl *blockListener) dispatchToConsumers(consumers []*blockUpdateConsumer, u
 	}
 }
 
-func (bl *blockListener) checkStartedLocked() {
+func (bl *blockListener) checkStartedLocked(ctx context.Context) {
 	if bl.listenLoopDone == nil {
 		bl.listenLoopDone = make(chan struct{})
 		go bl.listenLoop()
 	}
+
+	bl.blockTillBlockFilterEstablished(ctx)
```

Comment:
why would we do this for every consumer? If the `listenLoop` is already running before. I think we only need this once

Reply:
This would block adding new consumers

Reply:
within bl.blocktillblockfilterestablished, it checks if the channel is closed, if the loop is already running, the channel will be closed, this won't block

Reply:
We do if we put it inside the if block

Reply:
Makes sense, I'll test it and shift it inside the if block if nothing breaks

Reply:
what @rodion-lim-partior has done here makes sense to me. Using a mutex to ensure that there will be only 1 instance of `listenLoop` to close the `signal` channel while ensuring all other go routines wait until the `signal` channel is closed matches the behavior we discussed.

Reply:
Okay, so when I wrote this comment it was in conjunction with the call to `checkStartedLocked` still being locked by `bl.mux.Lock()` but yes if we are now removing it we would need a lock at this point